src/test/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSparkUnitTest.java

```diff
@@ -0,0 +1,95 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceWindowFunctions;
+import org.broadinstitute.hellbender.engine.spark.SparkContextFactory;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Unit tests for FindBadGenomicKmersSpark
+ */
+public class FindBadGenomicKmersSparkUnitTest extends BaseTest {
+
+    private static final int KMER_SIZE = SVConstants.KMER_SIZE;
+    private static final String REFERENCE_FILE_NAME = hg19MiniReference;
+
+    @Test(groups = "spark")
+    public void badKmersTest() throws IOException {
+        final File refSequenceFile = File.createTempFile("genomic_sequence", "txt");
+        refSequenceFile.deleteOnExit();
```

Comment:
Also, given that `createTempFile()` guarantees that names will be uniquified within the temp directory for each call, names of indices are generated using the basename of the file they index, and index formats have well-defined extensions, the prospect of a name collision with an index in the temp dir marked for deletion on exit is pretty academic (and has never been an issue, unlike the problem of people forgetting to clean these things up).